### PR TITLE
v2.8.1

### PR DESCRIPTION
**Pull request by FedorUporov**
Fix issues:
- https://github.com/electricimp/Builder/issues/66
- https://github.com/electricimp/Builder/issues/68

Notes:
- The sample-17 test will pass after the PR will be merged to develop@EI.
- Later, the githubPath const in sample-17 could be switched to master branch.

### CODE REVIEW

File: spec/Machine/sample-17.spec.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// http://opensource.org/licenses/MIT
+
+'use strict';
+
+require('jasmine-expect');
+
+const eol = require('eol');
+const init = require('./init')('main');
+
+const githubPath = 'github:electricimp/Builder/spec/fixtures/sample-17/lib/lib.nut@develop';
```
Comment (betzrhodes) [inline spec/Machine/sample-17.spec.js]: Note from emails: This branch should update to master before release.

Comment (betzrhodes) [reply spec/Machine/sample-17.spec.js]: flagging this line for myself

**Comment by betzrhodes** [comment]
Windows tests are failing locally - 
1) \spec\HttpReader.spec.js:27:20 -  HttpReader should read sample#1 from githubusercontenet.com (http)
2) \spec\HttpReader.spec.js:33:20 - HttpReader should read sample#1 from githubusercontext.com (https)
3) \spec\Machine\sample-8.spec.js:28:20 - Machine should run sample #8
4) Test timed out - Machine should print included source duplicate warning by default

**Comment by ppetrosh** [comment]
Once it's landed on develop we should be able to see why it's failing from our side. Thanks, Betsy. Please keep it in this state for a while.

**Comment by betzrhodes** [comment]
Sure I don't know what the end of line setting are on the local windows machine I just grabbed for testing.

**Comment by betzrhodes** [comment]
Windows specs for the local machine here: 
HP Notebook AMD A6-7310 API with AMD Radeon R4 Graphics 2.00 Ghz 
4.00GB RAM
64-bit OS
Windows 10, v1803, OS build 17134.345
node v10.15.3, npm v6.4.1

**Comment by ppetrosh** [comment]
Thanks, @betzrhodes! That's helpful.

**Pull request merged**